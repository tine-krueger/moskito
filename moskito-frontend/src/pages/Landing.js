import { Link } from 'react-router-dom'
import styled from 'styled-components/macro'
import FrontHeadline from '../components/FrontHeadline/FrontHeadline'
import Button from '../components/Button/Button';
import Logo from "../components/Logo/Logo";

export default function Landing() {

    return (
        <FrontGrid>
            <div>
                <Logo/>
                <FrontHeadline>Camping back to the roots</FrontHeadline>
            </div>
            <UserButtons>
                <Link to={'/login'}>
                    <Button>Login </Button> 
                </Link>
                <Link to={'/signin'}>
                    <Button>SignUp </Button>
                </Link>    
            </UserButtons>
        </FrontGrid>
    )
}

const FrontGrid = styled.div`
    display: grid;
    grid-template-rows: 2fr 3fr;
`
const UserButtons = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    Button {
        margin: .5em;
        width: 10em;
    }
`
