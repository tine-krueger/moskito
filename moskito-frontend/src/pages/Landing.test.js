import { screen } from '@testing-library/react'
import 'jest-styled-components'
import userEvent from '@testing-library/user-event'
import Landing from './Landing'
import renderWithRouter from '../testSetup/setupTests'

describe('Landing', () => {
    it('renders correctly', () => {
        const { container } = renderWithRouter(<Landing/>)
        expect (container).toMatchSnapshot()
    })

    it('has a headline and two buttons', () => {
        const { getByRole, getByTestId } = renderWithRouter(<Landing/>)
        expect(getByRole('button', {name: /Login/i})).toBeInTheDocument()
        expect(getByRole('button', {name: /Signup/i})).toBeInTheDocument()
        expect(getByRole('heading', {name: /Camping back to the roots/i})).toBeInTheDocument()
        expect(getByTestId('logo')).toBeInTheDocument()
    })

    it('directs to sign-up-page after click on signup-button', () => {
        const { history } = renderWithRouter(<Landing/>)
        userEvent.click(screen.getByText(/SignUp/i))
        expect(history.location.pathname).toBe('/signup') 
    })

    it('directs to login-page after click on login-button', () => {
        const { history } = renderWithRouter(<Landing/>)
        userEvent.click(screen.getByText(/Login/i))
        expect(history.location.pathname).toBe('/login') 
    })

})