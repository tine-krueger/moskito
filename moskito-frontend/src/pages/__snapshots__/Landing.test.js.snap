// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders correctly 1`] = `
.c3 {
  font-family: 'Hind Madurai';
  font-size: 1.5em;
  text-align: center;
  margin-top: 1em;
}

.c5 {
  padding: 1em 2em;
  font-family: 'Arvo';
  font-size: 1.1rem;
  border-radius: 10px;
  width: auto;
  border: none;
  background: var(--background);
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
  color: var(--headline);
  text-transform: uppercase;
  margin: 1em auto 3em auto;
}

.c5:hover {
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
  outline: none;
}

.c1 {
  padding-top: 2em;
  text-align: center;
}

.c2 {
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
  border-radius: 20px;
  padding: 1em 2em;
  width: 85%;
}

.c0 {
  display: grid;
  grid-template-rows: 2fr 3fr;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 Button {
  margin: .5em;
  width: 10em;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
        data-testid="logo"
      >
        <img
          alt="Moskito Logo"
          class="c2"
          src="moskitoLogo.svg"
        />
      </div>
      <h1
        class="c3"
      >
        Camping back to the roots
      </h1>
    </div>
    <div
      class="c4"
    >
      <a
        href="/login"
      >
        <button
          class="c5"
        >
          Login 
        </button>
      </a>
      <a
        href="/signup"
      >
        <button
          class="c5"
        >
          SignUp 
        </button>
      </a>
    </div>
  </div>
</div>
`;
